src/docfx/restore/index/DependencyGitPool.cs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.Docs.Build
+{
+    internal class DependencyGitPool
+    {
+        private Dictionary<(string remote, string branch, string commit, LockType lockType), (string path, DependencyGit git, bool released)> _acquiredGits = new Dictionary<(string remote, string branch, string commit, LockType lockType), (string path, DependencyGit git, bool released)>();
```

Comment:
can this dictionary be `readonly IReadOnlyDictionary`, to separate mutating operations like `AcquireXXX` from non mutating (thread safe) operations like `GetXXX` and `ReleaseXXX`. Like builder patterns used in other places, mutating operations returns an immutable instance.

Reply:
great idea, try try